modules/diskq/qdisk.h

```diff
@@ -54,6 +54,8 @@ gint64 qdisk_get_empty_space(QDisk *self);
 gboolean qdisk_push_tail(QDisk *self, GString *record);
 gboolean qdisk_pop_head(QDisk *self, GString *record);
 gboolean qdisk_remove_head(QDisk *self);
+gint64 qdisk_get_next_tail_position(QDisk *self);
+gint64 qdisk_get_head_position(QDisk *self);
```

Comment:
I think publishing positions for anything other than tests is not a good idea, but currently we do that to track things outside, in memory-only queue segments.

It looks better and a little bit simpler with these new methods, but ideally we should not need these at all in my opinion (that was the reason why the bug was actually introduced, because we tried to be clever about the internals of `QDisk`, but something was changed inside, therefore something else broke outside).

We won't have the time to refactor/fix everything this time, but I think we should mention the above in the commit message, so we won't forget it completely.

Reply:
Thanks, I agree. I added some comments about it in the commit msg:
```
    qdisk: add and use get_next_tail_position() and get_head_position()                         

    We need these info in the reliable disk-buffer, because it is matching                      
    its memory queue's message to the qdisk's messages by their position                        
    in the qdisk.                                                                               

    Currently we guess these positions on the caller side, based on the                         
    write_head's and read_head's position.      

    This guess is not always right, because for example push_tail() can
    change the write_head before writing to the disk.                                           

    ------------------------------------------------------------------------                    

    This commit introduces a new interface around QDisk, where we do not                        
    make the internal read-, write- and backlog-head public, but give                           
    functions closer to a queue implementation (head and tail).                                 

    This is still not the ideal interface. Ideally QDisk should not give                        
    any info about the records position on the disk, as it is still an                          
    implementation detail, somehow the callor site should keep track of the
    messages' location, with an index array, or something like that.

    Unfortunately we cannot refactor that much within the scope of this PR,                     
    that will be done later.                                                                    

    ------------------------------------------------------------------------                    

    The other problem is that we still have the 2 not so perfect interface
    in QDisk simultaneously. The reason behind not removing the old one is
    that we do not have a good place for get_backlog_head() in the new                          
    interface.                                                                                  

    Though, if we say, that QDisk has a backlog-like logic inside                               
    - which it has - then the moving of the backlog_head should be done
    in QDisk, not with public getter and setter functions.                                      

    If the backlog_head modification happens solely in QDisk, we will not                       
    need any public function to do so, so we can completely remove the old                      
    interface and use the new one.                                                              

    We think, that the two mentioned problem can be fixed in one go, and we                     
    plan to do it later.

    Signed-off-by: Attila Szakacs <attila.szakacs@oneidentity.com>
```